### PR TITLE
[JS] Topic: Regular Expressions

### LINKED ISSUE #2142: [JS] Issue: Regular Expressions
### Description

Introducing the topic of Regular Expressions and their significance in JavaScript.

<b>Mode: </b> Documentation

## Directory
`JavaScript/Topics/13.Regular_Expressions`

### Domain

JavaScript

### Type of Contribution

- [ ] Documentation
- [ ] Video
- [ ] Audio

### Code of Conduct

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.

### PR DESCRIPTION
**Pull request by subhendudash02**
<hr>

## Description 📜
Added the documentation regarding the regular expressions in JavaScript in the form of a markdown file.
Added the images in a separate sub-folder.

Fixes #2142
<hr>

## Type of change 📝

- [ ] Audio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Video (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Documentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [x] JavaScript

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

<hr>

### CODE REVIEW

**Review by sushovanb02 — APPROVED** [review]

**Comment by s4-2** [comment]
@subhendudash02 Very well formated and detailed information. Well done:)

**Review by s4-2 — APPROVED** [review]

**Comment by subhendudash02** [comment]
> @subhendudash02 Very well formated and detailed information. Well done:)

😊Glad to know @s4-2 !! Thank you so much for the review!

File: JavaScript/Topics/Regular_Expressions/README.md [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+Regular expressions can also be represented using `RegExp()` function.
+
+<b>Syntax: </b> `RegExp(regularExp, "modifier")`
+
+For example, 
+
+``` JS
+let reg = RegExp("x+", "g");
+```
```
Comment (paramsiddharth) [inline JavaScript/Topics/Regular_Expressions/README.md:22]: Do this instead of that.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Review by paramsiddharth — APPROVED** [review]

**Review by dakshsethi — APPROVED** [review]
Looks good to me!! 🔥🔥🔥

**Review by paramsiddharth — APPROVED** [review]

**Pull request merged**